src/models/history/tree-manager.ts

```diff
@@ -336,7 +336,7 @@ export const TreeManager = types
 
 }))
 .actions((self) => ({
-  mirrorHistoryFromFirestore(user: UserModelType, firestore: Firestore) {
+  async mirrorHistoryFromFirestore(user: UserModelType, firestore: Firestore) {
```

Comment:
It'd be good to add a comment here:
> Using async actions is generally not a good idea, because the changes will not be grouped into a single action for any action tracking middleware. However in this case we aren't recording the actions of the TreeManager.